src/sentry/issues/occurrence_consumer.py

```diff
@@ -356,3 +362,49 @@ def _process_message(
             txn.set_tag("result", "error")
             raise InvalidEventPayloadError(e)
     return None
+
+
+def _process_batch(_occurrence_worker, message: Message[ValuesBatch[KafkaPayload]]):
+    """
+    Receives batches of occurrences. This function will take the batch
+    and group them together by fingerprint (ensuring order is preserved) and
+    execute each group using a ThreadPoolWorker.
+
+    By batching we're able to process occurrences in parallel while guaranteeing
+    that no occurrences are processed out of order per group.
+    """
+    batch = message.payload
+
+    occcurrence_mapping: Mapping[str, list[CheckinItem]] = defaultdict(list)
+
+    for item in batch:
+        assert isinstance(item, BrokerValue)
+
+        try:
+            payload = json.loads(item.payload.value, use_rapid_json=True)
+        except Exception:
+            logger.exception("Failed to unpack message payload")
+            continue
+        occcurrence_mapping[item.partition.index].append(payload)
```

Comment:
If you use the actual fingerprint we can create more groups and execute more things in parallel.

Reply:
@evanpurkhiser how is that different than this exactly? Wouldn't they be the same?

Reply:
no because one consumer is only going to be responsible for `n` number of partitions, so you'll only ever have `n` threads processing all the various fingerprints in that partition.

If you group by fingerprint instead, you'll have lots of groups with single fingerprints that can all be processed in parallel.